### ISSUE TITLE
Add registered contrastive logits difference step function

### ISSUE THREAD

**Issue by gsarti** [inseq-team/inseq#147]
## Motivation

Contrastive attributions are currently supported thanks to custom attributed targets (see #138). The current definition of the contrastive attribution custom function can be found [here](https://github.com/inseq-team/inseq/blob/9e88a9ebf3788043ec6cc0c6fa334f594a29b154/tests/attr/feat/test_feature_attribution.py#L27).

The current implementation is problematic for `integrated_gradients` and similar methods using multiple approximation steps since the contrastive forward uses static ids instead of embeddings obtained as steps between the original contrastive input and a baseline. Moreover, the current implementation allows only for granular token-based comparisons proposed in the original work by [Yin and Neubig (2022)](https://arxiv.org/abs/2202.10419), but comparing spans of different lengths could also be desirable.

Given this will add further complexity to the custom attributed function, and given the interest in such an application, it would be ideal to include a pre-registered version of the contrastive attribution step function inside Inseq to enable easy and quick usage.

## Design

**Step function name:** `contrast_probs_diff`, since the contrastive comparison is done by taking the difference of output probabilities between a regular and a contrastive example.

**Extra arguments**:

- `contrast_input`: required, can be either an input text, a sequence of ids or embeddings for the contrastive example. The function will handle the formatting to match the original input.
- `input_start_span_ids`, `contrast_start_span_ids`: Two lists containing initial ids for ever span in the input and the contrast that we want to consider as single units for attribution purposes (e.g. `[0, 2, 5]` for `input_start_span_ids` says By default `None`, set to `list(range(len(input_ids)))` and `list(range(len(contrast_ids)))` respectively (i.e. every token is treated separately, as in normal feature attribution). 

**Notes for span ids**: 
- Must verify that all ids are valid given string length, and that the two lists have same length to ensure contrastive comparison for every span step.
- The step function is called at every token, but we consider multi-token spans. If the current token is an start span token for the input, we compute the product of probabilities for all tokens in the current input span, the corresponding product for all tokens in the current contrast span and output their diff. If it's not a start token (i.e. its id in the span is not included in `input_start_span_ids`) 0 is returned.

**Aggregation**:

The default `abs_max` function used for span aggregation (`span_aggregate`) of attributions will return the attributions for the first token of every span (see description in the previous section) if the same spans are used with a `ContiguousSpanAggregator`. The `aggregate_map` for the step score should also be set to `abs_max` upon registration.
